compiler/rustc_mir/src/util/pretty.rs

```diff
@@ -408,6 +409,33 @@ impl ExtraComments<'tcx> {
     }
 }
 
+fn use_verbose(ty: &&TyS<'tcx>) -> bool {
+    match ty.kind() {
+        ty::Int(_) | ty::Uint(_) | ty::Bool | ty::Char | ty::Float(_) => false,
+        // Unit type
+        ty::Tuple(g_args) if g_args.is_empty() => false,
+        ty::Tuple(g_args) => {
+            // could have used `try_fold` here but it seems a bit silly that
+            // the accumulator is useless
+            let mut should_be_verbose = false;
+            for g_arg in g_args.iter() {
+                if match g_arg.unpack() {
+                    GenericArgKind::Type(ty) => use_verbose(&ty),
+                    GenericArgKind::Const(ty::Const { ty, val: _ }) => use_verbose(ty),
```

Comment:
Tuple fields must all be `GenericArgKind::Type`, so you can in fact use `expect_ty` on `g_arg` and do not have to handle the const/lifetime cases